quarkus/dist/src/main/content/bin/kc.bat

```diff
@@ -225,17 +222,15 @@ if "%PRINT_ENV%" == "true" (
 
 set START_SERVER=true
 
-if "!CONFIG_ARGS:%OPTIMIZED_OPTION%=!"=="!CONFIG_ARGS!" if "!CONFIG_ARGS:%BUILD_OPTION%=!"=="!CONFIG_ARGS!" if "!CONFIG_ARGS:%HELP_LONG_OPTION%=!"=="!CONFIG_ARGS!" if "%IS_HELP_SHORT%" == "false" (
+"%JAVA%" !JAVA_RUN_OPTS!
 
-    "%JAVA%" -Dkc.config.build-and-exit=true !JAVA_RUN_OPTS!
-
-    if not !errorlevel! == 0 (
-        set START_SERVER=false
-    )
-
-    set JAVA_RUN_OPTS=-Dkc.config.built=true !JAVA_RUN_OPTS!
+rem only exit code 10 means that implicit reaugmentation occurred and a relaunch is needed
+if not !errorlevel! == 10 (
+    set START_SERVER=false
 )
 
+set JAVA_RUN_OPTS=-Dkc.config.built=true !JAVA_RUN_OPTS!
+
 if "%START_SERVER%" == "true" (
     "%JAVA%" !JAVA_RUN_OPTS!
 )
```

Comment:
Just a nitpick:
```suggestion

"%JAVA%" !JAVA_RUN_OPTS!

rem only exit code 10 means that implicit reaugmentation occurred and a relaunch is needed
if not !errorlevel! == 10 (
    exit /B !errorlevel!
)

set JAVA_RUN_OPTS=-Dkc.config.built=true !JAVA_RUN_OPTS!

"%JAVA%" !JAVA_RUN_OPTS!
exit /B !errorlevel!

```

I guess that this mirrors kc.sh's logic better and it avoids any difference caused by setting the flag earlier. It also makes exit behavior explicit on Windows.

Reply:
I had some difficulties to mark the intended range of code for suggestion, so just to be clear, here's what I'm suggesting:
<img width="2081" height="528" alt="image" src="https://github.com/user-attachments/assets/31524107-52f5-4fbc-8a6f-3be3c6f8cc1f" />